nix/pkgs/status-go/library/default.nix

```diff
@@ -8,10 +8,30 @@
 
 let
   optionalString = pkgs.lib.optionalString;
+  codexVersion = "v0.0.28";
+  arch =
+    if stdenv.hostPlatform.isx86_64 then "amd64"
+    else if stdenv.hostPlatform.isAarch64 then "arm64"
+    else stdenv.hostPlatform.arch;
+  os = if stdenv.isDarwin then "macos" else "Linux";
+  hash = 
+    if stdenv.hostPlatform.isDarwin 
+    # nix store prefetch-file --json --unpack https://github.com/logos-storage/logos-storage-go-bindings/releases/download/${codexVersion}/codex-macos-arm64.zip | jq -r .hash
+    then "sha256-GcerkH8izZ5QHG5ARNNrM1fktaeBKjF6AGNsA6vxVj0="
+    # nix store prefetch-file --json --unpack https://github.com/logos-storage/logos-storage-go-bindings/releases/download/${codexVersion}/codex-Linux-amd64.zip | jq -r .hash
+    else "sha256-sYhbgBN0LNA7YhmBigPwo1h34QTADTxFGjO8QAw8m18=";
+
+  # Pre-fetch libcodex to avoid network during build
+  codexLib = pkgs.fetchzip {
+    url = "https://github.com/logos-storage/logos-storage-go-bindings/releases/download/${codexVersion}/codex-${os}-${arch}.zip";
+    hash = hash;
+    stripRoot = false;
+  };
```

Comment:
I'm not a Nix pro, but I feel like this should be defined as a separate derivation, not in `status-go/library`.